### PR TITLE
LIVE-3475: Enable Solana in swap

### PR DESCRIPTION
**Pull request by jnicoulaud-ledger**
### 📝 Description

Enable Solana support in swap.

### ❓ Context

- **Impacted projects**: LLD, LLM
- **Linked resource(s)**:
    - https://ledgerhq.atlassian.net/browse/LIVE-3475
    - https://ledgerhq.atlassian.net/browse/BACK-3827

### ✅ Checklist

- [ ] **Test coverage**
- [x] **Atomic delivery**
- [x] **No breaking changes**

### 📸 Demo

TODO

### 🚀 Expectations to reach

Solana from/to swaps works with Changelly and CIC.

### CODE REVIEW

**Review by haammar-ledger — COMMENTED** [review]
LGTM, but I have very little knowledge of the exchange feature.
Maybe @juan-cortes you could review too.

**Review by juan-cortes — APPROVED** [review]
lgtm

File: libs/ledger-live-common/src/families/solana/exchange.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { bip32asBuffer } from "@ledgerhq/hw-app-btc/bip32";
```
Comment (haammar-ledger) [inline libs/ledger-live-common/src/families/solana/exchange.ts:1]: Really not a good pattern here that `solana` has a dependency to the `hw-app-btc` lib, but I can see other families are already doing it so... not a blocker here.

**Review by haammar-ledger — APPROVED** [review]
Pending QA before merging.

File: libs/ledger-live-common/src/families/solana/exchange.ts (hunk 1 above)
Comment (jnicoulaud-ledger) [reply libs/ledger-live-common/src/families/solana/exchange.ts:1]: Yes I preferred to keep it consistent :angel:

**Comment by jnicoulaud-ledger** [comment]
Current status: this is mergeable but swap crashes in solana coin app. I will complete the PR description whenever we can finally test it.

https://ledgerhq.atlassian.net/browse/BACK-3827 will be kept up to date.

**Comment by haammar-ledger** [comment]
@jnicoulaud-ledger There seems to be some small issue with the generated `exchange.ts` file: https://github.com/LedgerHQ/ledger-live/actions/runs/3524934384/jobs/6366484465#step:9:5152

**Comment by jnicoulaud-ledger** [comment]
@haammar-ledger thanks, fixed

**Comment by haammar-ledger** [comment]
@jnicoulaud-ledger Not sure why one test keeps failing on desktop, doesn't seem related to this PR indeed.
If it keeps failing you can ask for help on `#playwright` slack chan.

**Comment by haammar-ledger** [comment]
@jnicoulaud-ledger I believe there were a number of changes done on the CI recently, could you please rebase? Let's give it another shot.

**Review by gre — APPROVED** [review]

**Comment by jnicoulaud-ledger** [comment]
superseded by #2560

**Pull request closed without merging**